### PR TITLE
[test_scenes] Port longpathdash test case from Skia

### PR DESCRIPTION
**Pull request by armansito**
This is a single very long path that results in a large number of line segments when dashing is applied. This demonstrates rendering artifacts.

### CODE REVIEW

File: examples/scenes/src/test_scenes.rs [hunk 1]
```diff
@@ -21,26 +21,21 @@ macro_rules! scene {
             function: Box::new($name),
```
Comment (DJMcNab) [inline examples/scenes/src/test_scenes.rs:21]: I'd probably express this case (lines 16-22) in terms of the new addition

**Review by DJMcNab — COMMENTED** [review]
I've not had a chance to run this yet - can you include a screenshot?

**Comment by armansito** [comment]
> I've not had a chance to run this yet - can you include a screenshot?

Yes of course! 

This is the expected shape with butt caps:
Correct butt cap

This is the same butt cap render showing artifacts when it's zoomed out:
Butt cap - zoomed out artifacts

This is the same shape with round caps (which increases the line segment count) showing artifacts at a moderate zoom level:
Screenshot 2023-10-11 at 10 18 42 AM

The latter can be fixed by increasing the bump buffer sizes. I'm treating this as a pathological case with a very high memory requirement. The zoomed out case appears regardless of the size of the bump buffers so that's probably a separate issue.

File: examples/scenes/src/test_scenes.rs (hunk 1 above)
Comment (armansito) [reply examples/scenes/src/test_scenes.rs]: Done.

**Review by raphlinus — APPROVED** [review]
Thanks, and also for the cleanup of the scene creation macros. It will be useful to have more examples that stress the buffer sizes, as we make our way through the dynamic memory question.

**Pull request merged**